File: app/src/main/java/com/iffy/mianshi/basic/fanxing/basic/TestFanXing.java
package com.iffy.mianshi.basic.fanxing.basic;

import android.os.Build;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.function.Consumer;

/**
 * author : iffy
 * time   : 2020/03/12
 */

public class TestFanXing {
    public static void main(String[] args) {
        test();
    }

    public static void test(){
        System.out.println("Start");
        ArrayList list = new ArrayList();
        list.add(123);
        list.add(456);
        list.add("123");//泛型就是为了解决数据存储的安全性
        Iterator iterator = list.iterator();
        while (iterator.hasNext()){
            System.out.println((Integer)iterator.next());
        }
        System.out.println("End");
    }

}

File: app/src/main/java/com/iffy/mianshi/basic/fanxing/basic/TestFanXing.kt
package com.iffy.mianshi.basic.fanxing.basic

fun main() {
    FanXing.set("我是泛型方法")
    FanXingLei<String>().sayHello("你好泛型类")

    val fanXinTest = FanXingLeiB(10)
    println(fanXinTest.value)
    fanXinTest.value = 9
    println(fanXinTest.value)
    println(fanXinTest.value)

    //通配符
    var list = ArrayList<String>()
    var listB = ArrayList<Int>()
    test(list)
    test(listB)

}

fun test(list: ArrayList<*>) {

}

